plugin/inlay_hint.py

```diff
@@ -0,0 +1,155 @@
+                background-color: color(var(--foreground) alpha(0.08));
+                border-radius: 4px;
+                margin-left: {margin_left};
+                margin-right: {margin_right};
+                padding: 0.05em 4px;
+                font-size: 0.9em;
+                font-family: {font};
+            }}
+
+            .inlay-hint a {{
+                color: color(var(--foreground) alpha(0.6));
+                text-decoration: none;
+            }}
+        </style>
+        <div class="inlay-hint" title="{tooltip}">
+            {label}
+        </div>
+    </body>
+    """.format(
+        margin_left=margin_left,
+        margin_right=margin_right,
+        tooltip=tooltip,
+        font=font,
+        label=label
+    )
+    return html
+
+
+def format_inlay_hint_tooltip(tooltip: Optional[Union[str, MarkupContent]]) -> str:
+    if isinstance(tooltip, str):
+        # Sublime Text doesn't support HTML for tooltips
+        # so we strip the HTML tags from the tooltip
+        return tooltip
+    if isinstance(tooltip, dict):  # MarkupContent
+        return tooltip.get('value') or ""
+    return ""
+
+
+def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    result = ""
+    can_resolve_inlay_hint = session.has_capability('inlayHintProvider.resolveProvider')
+    label = inlay_hint['label']  # type: ignore
+    is_clickable = bool(inlay_hint.get('textEdits')) or can_resolve_inlay_hint
+    if isinstance(label, str):
+        inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
+            'session_name': session.config.name,
+            'inlay_hint': inlay_hint,
+            'phantom_uuid': phantom_uuid
+        })
+        if is_clickable:
+            result += '<a href="{command}">'.format(command=inlay_hint_click_command)
+        result += html.escape(label)
+        if is_clickable:
+            result += "</a>"
+        return result
+
+    for label_part in label:
+        value = ""
+        inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
+            'session_name': session.config.name,
+            'inlay_hint': inlay_hint,
+            'phantom_uuid': phantom_uuid,
+            'label_part': label_part
+        })
+        is_clickable = is_clickable or bool(label_part.get('command'))
+        if is_clickable:
+            value += '<a href="{command}">'.format(command=inlay_hint_click_command)
+        value += html.escape(label_part.get('value') or "")
+        if is_clickable:
+            value += "</a>"
+        # InlayHintLabelPart.location is not supported
+        result += "<div title=\"{tooltip}\">{value}</div>".format(
+            tooltip=format_inlay_hint_tooltip(label_part.get("tooltip")),
+            value=value
+        )
+    return result
+
+
+def inlay_hint_to_phantom(view: sublime.View, inlay_hint: InlayHint, session: Session) -> sublime.Phantom:
```

Comment:
Can you move this function before the `get_inlay_hint_html` so that it's in an easier to read order?